### PR TITLE
Components suite logs

### LINKED ISSUE #938: [Feature Request]
Thanks so much for your interest in Dash!

Before posting an issue here, please check the Dash [community forum](https://community.plot.ly/c/dash) to see if the topic has already been discussed. The community forum is also great for implementation questions. When in doubt, please feel free to just post the issue here :)

**Is your feature request related to a problem? Please describe.**
after an investigation in recent DCC-CI setup, we identified a long-existing trouble maker, at run time, that the dash app loads `dash_core_components` folder in check-out source code, rather than the expected installed path in `venv`.

there is no logging trace at dash backend side to tell which package we are serving and from where. 

**Describe the solution you'd like**
add logging in python backend

### PR DESCRIPTION
**Pull request by byronz**
*Start with a description of this PR. Then edit the list below to the items that make sense for your PR scope, and check off the boxes as you go!*

## Contributor Checklist

- [x ] I have broken down my PR scope into the following TODO tasks
   -  [x] fix
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))

### optionals

- [x] I have added entry in the `CHANGELOG.md`

this closes #938

### CODE REVIEW

File: dash/dash.py [hunk 1]
```diff
@@ -677,6 +677,14 @@ def serve_component_suites(self, package_name, path_in_package_dist):
             'map': 'application/json'
         })[path_in_package_dist.split('.')[-1]]
 
+        loader = pkgutil.get_loader(package_name)
+        self.logger.info(
+            "serving package %s resource %s => %s",
+            package_name,
+            path_in_package_dist,
+            loader.path if hasattr(loader, 'path') else loader,
```
Comment (byronz) [inline dash/dash.py]: python 2 has different `loader` protocol, so we will see the path only in python3

**Comment by byronz** [comment]
the logs will appear in CI logs only if the test case failed

File: dash/dash.py (hunk 1 above)
Comment (alexcjohnson) [reply dash/dash.py]: [elsewhere](https://github.com/plotly/dash/blob/a42b712e1a88d17400726b5b28ef20a36e826d21/dash/development/base_component.py#L22) we just reference `sys.modules[package_name]` - can we just get the package there and grab its `__path__`?

File: dash/dash.py [hunk 2]
```diff
@@ -678,7 +678,9 @@ def serve_component_suites(self, package_name, path_in_package_dist):
         })[path_in_package_dist.split('.')[-1]]
 
         loader = pkgutil.get_loader(package_name)
-        self.logger.info(
+        msg = getattr(
+            self.logger, 'info' if self.config.serve_locally else 'debug')
```
Comment (alexcjohnson) [inline dash/dash.py]: do we actually get here if we're not serving locally?

Comment (byronz) [reply dash/dash.py]: Interesting question. the flag that I really want is the `debug` argument but in this function `self.config` is easier to access. 

the logic behind using serve_locally is from here https://github.com/plotly/dash/blob/a42b712e1a88d17400726b5b28ef20a36e826d21/dash/dash.py#L1557-L1560

it's False by default and set to True when debug is True

Comment (byronz) [reply dash/dash.py]: I tested with debug=True/False,  the logging level get changed as expected

File: dash/dash.py (hunk 1 above)
Comment (byronz) [reply dash/dash.py]: tried in python 2.7, I think it's basically the same, as the protocol is at the lower level, so you won't even get `__path__` from the loader object

Comment (alexcjohnson) [reply dash/dash.py]: Ah I see, you're right, `module.__path__` and `module.__file__` *exist* but they aren't really paths.

**Pull request merged**